core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -218,6 +222,40 @@ public static void updateTableInfo(TableInfo tableInfo) {
     }
   }
 
+  public static CarbonTable buildTable(
+      String tablePath,
+      String tableName,
+      boolean hasFilter) throws IOException {
```

Comment:
There should not be any filter related flags in Carbon Table API, if you require this check then move to utilities. It should only has buildTable method not `hasFilter` flag.

Reply:
ok.done